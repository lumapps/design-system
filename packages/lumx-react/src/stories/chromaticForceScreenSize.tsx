/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
