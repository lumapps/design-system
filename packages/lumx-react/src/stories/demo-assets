../../../site-demo/static/demo-assets