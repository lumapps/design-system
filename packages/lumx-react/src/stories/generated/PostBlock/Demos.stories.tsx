/**
 * File generated when storybook is started. Do not edit directly!
 */
export default { title: 'LumX components/post-block/Demos' };

export { App as Default } from './default';
