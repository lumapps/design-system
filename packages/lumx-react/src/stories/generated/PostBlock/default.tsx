/home/gcornut/git/dup-design-system/packages/site-demo/content/product/components/post-block/react/default.tsx