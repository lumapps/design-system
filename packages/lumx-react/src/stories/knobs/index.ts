export * from './image';
export * from './lorem';
