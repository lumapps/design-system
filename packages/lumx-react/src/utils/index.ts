/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
