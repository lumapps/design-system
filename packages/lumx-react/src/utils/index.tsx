export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
