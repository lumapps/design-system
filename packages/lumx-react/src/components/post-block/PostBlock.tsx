import React, { forwardRef, ReactNode } from 'react';

import classNames from 'classnames';

import isObject from 'lodash/isObject';

import { Orientation, Theme, Thumbnail, ThumbnailProps, ThumbnailVariant } from '@lumx/react';

import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';

/**
 * Defines the props of the component.
 */
export interface PostBlockProps extends GenericProps {
    /** Action toolbar content. */
    actions?: ReactNode;
    /** Attachment content. */
    attachments?: ReactNode;
    /** Author content. */
    author?: ReactNode;
    /** Metadata content. */
    meta?: ReactNode;
    /** Orientation. */
    orientation?: Orientation;
    /** Tag content. */
    tags?: ReactNode;
    /** Content (string, or sanitized html). */
    text?: string | { __html: string };
    /** Theme adapting the component to light or dark background. */
    theme?: Theme;
    /** Thumbnail. */
    thumbnailProps?: ThumbnailProps;
    /** Title. */
    title: string;
    /** On click callback. */
    onClick?(): void;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'PostBlock';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * Component default props.
 */
const DEFAULT_PROPS: Partial<PostBlockProps> = {
    orientation: Orientation.horizontal,
    theme: Theme.light,
};

/**
 * PostBlock component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const PostBlock: Comp<PostBlockProps, HTMLDivElement> = forwardRef((props, ref) => {
    const {
        actions,
        attachments,
        author,
        className,
        meta,
        onClick,
        orientation,
        tags,
        text,
        theme,
        thumbnailProps,
        title,
        ...forwardedProps
    } = props;

    return (
        <div
            ref={ref}
            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, orientation, theme }))}
            {...forwardedProps}
        >
            {thumbnailProps && (
                <div className={`${CLASSNAME}__thumbnail`}>
                    <Thumbnail {...thumbnailProps} theme={theme} variant={ThumbnailVariant.rounded} />
                </div>
            )}

            <div className={`${CLASSNAME}__wrapper`}>
                {author && <div className={`${CLASSNAME}__author`}>{author}</div>}

                {title && (
                    <button type="button" className={`${CLASSNAME}__title`} onClick={onClick}>
                        {title}
                    </button>
                )}

                {meta && <span className={`${CLASSNAME}__meta`}>{meta}</span>}

                {isObject(text) && text.__html ? (
                    // eslint-disable-next-line react/no-danger
                    <p dangerouslySetInnerHTML={text} className={`${CLASSNAME}__text`} />
                ) : (
                    <p className={`${CLASSNAME}__text`}>{text}</p>
                )}

                {attachments && <div className={`${CLASSNAME}__attachments`}>{attachments}</div>}
                {(tags || actions) && (
                    <div className={`${CLASSNAME}__toolbar`}>
                        {tags && <div className={`${CLASSNAME}__tags`}>{tags}</div>}
                        {actions && <div className={`${CLASSNAME}__actions`}>{actions}</div>}
                    </div>
                )}
            </div>
        </div>
    );
});
PostBlock.displayName = COMPONENT_NAME;
PostBlock.className = CLASSNAME;
PostBlock.defaultProps = DEFAULT_PROPS;
