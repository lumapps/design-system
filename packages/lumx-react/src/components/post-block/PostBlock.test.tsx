import React from 'react';
import { mount, shallow } from 'enzyme';
import 'jest-enzyme';
import { commonTestsSuite, itShouldRenderStories } from '@lumx/react/testing/utils';

import { PostBlock, PostBlockProps } from './PostBlock';
import * as stories from '../../stories/generated/PostBlock/Demos.stories';

const CLASSNAME = PostBlock.className as string;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = (props: Partial<PostBlockProps> = {}, shallowRendering = true) => {
    const renderer: any = shallowRendering ? shallow : mount;
    const wrapper: any = renderer(<PostBlock {...(props as any)} />);
    return { props, wrapper };
};

describe(`<${PostBlock.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        itShouldRenderStories(stories, PostBlock);
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper', prop: 'wrapper' }, { className: CLASSNAME });
});
