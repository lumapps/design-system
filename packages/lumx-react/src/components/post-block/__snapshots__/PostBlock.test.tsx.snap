// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostBlock> Snapshots and structure should render story 'Default' 1`] = `
<div
  className="lumx-post-block lumx-post-block--orientation-horizontal lumx-post-block--theme-light"
>
  <div
    className="lumx-post-block__thumbnail"
  >
    <Thumbnail
      alt="Landscape"
      fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
      image="/demo-assets/landscape1.jpg"
      loading="lazy"
      theme="light"
      variant="rounded"
    />
  </div>
  <div
    className="lumx-post-block__wrapper"
  >
    <div
      className="lumx-post-block__author"
    >
      <UserBlock
        avatarProps={
          Object {
            "alt": "Avatar",
            "image": "/demo-assets/avatar2.jpg",
          }
        }
        name="Matthias Manoukian"
        orientation="horizontal"
        size="s"
        theme="light"
      />
    </div>
    <button
      className="lumx-post-block__title"
      type="button"
    >
      Annual Bonus Payments
    </button>
    <span
      className="lumx-post-block__meta"
    >
      1 day ago in community's name
    </span>
    <p
      className="lumx-post-block__text"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec rhoncus libero aliquet pharetra luctus. Fusce nisl turpis, posuere ac tellus at, euismod vulputate libero...",
        }
      }
    />
    <div
      className="lumx-post-block__attachments"
    >
      <UserBlock
        avatarProps={
          Object {
            "alt": "Avatar",
            "image": "/demo-assets/persona.png",
          }
        }
        fields={
          Array [
            "Head of Design",
            "Lyon",
          ]
        }
        name="Matthias Manoukian"
        orientation="vertical"
        size="l"
        theme="light"
      />
    </div>
    <div
      className="lumx-post-block__toolbar"
    >
      <div
        className="lumx-post-block__tags"
      >
        <ChipGroup
          align="left"
        >
          <Chip
            size="s"
            theme="light"
          >
            tag1
          </Chip>
          <Chip
            size="s"
            theme="light"
          >
            tag2
          </Chip>
          <Chip
            size="s"
            theme="light"
          >
            tag3
          </Chip>
        </ChipGroup>
      </div>
      <div
        className="lumx-post-block__actions"
      >
        <FlexBox
          orientation="horizontal"
        >
          <Button
            emphasis="low"
            leftIcon="M9,22A1,1 0 0,1 8,21V18H4A2,2 0 0,1 2,16V4C2,2.89 2.9,2 4,2H20A2,2 0 0,1 22,4V16A2,2 0 0,1 20,18H13.9L10.2,21.71C10,21.9 9.75,22 9.5,22V22H9Z"
            size="s"
            theme="light"
          >
            <span>
              16
            </span>
          </Button>
          <Button
            emphasis="low"
            leftIcon="M12,21.35L10.55,20.03C5.4,15.36 2,12.27 2,8.5C2,5.41 4.42,3 7.5,3C9.24,3 10.91,3.81 12,5.08C13.09,3.81 14.76,3 16.5,3C19.58,3 22,5.41 22,8.5C22,12.27 18.6,15.36 13.45,20.03L12,21.35Z"
            size="s"
            theme="light"
          >
            <span>
              8
            </span>
          </Button>
        </FlexBox>
      </div>
    </div>
  </div>
</div>
`;
